packages/compiler/test/tsx/escape.ts

```diff
@@ -43,4 +43,41 @@ export default function __AstroComponent_(_props: Record<string, any>): any {}\n
   assert.snapshot(code, output, 'expected code to match snapshot');
 });
 
+test('does not escape tag opening unnecessarily', async () => {
+  const input = `<div></div>
+<div`;
+  const output = `${TSXPrefix}<Fragment>
+<div></div>
+<div
+</Fragment>
+export default function __AstroComponent_(_props: Record<string, any>): any {}\n`;
+  const { code } = await convertToTSX(input, { sourcemap: 'external' });
+  assert.snapshot(code, output, 'expected code to match snapshot');
+});
+
+test('does not escape tag opening unnecessarily II', async () => {
+  const input = `<div>
+<div
+</div>
+`;
+  const output = `${TSXPrefix}<Fragment>
+<div>
+<div div {...{"<":true}}>
```

Comment:
This is what I mean by it not being perfect. Ideally, it would just be outputted as written, but the compiler thinks that those are attributes. This is very wonky looking, but as far as I could test, in the editor it's not a problem, TSX just think there's some text further on and an expression.

It might be possible to fix this, because the JS output actually does support this strangely enough. But it's perhaps a bit more complicated in TSX due to all the escaping required, not sure.